File: pyjl/tests_new/join_test.py

class Hello:
    def why(self):
        return "ola"

if __name__ == "__main__":
    # a = "a"
    # b = "ab"
    # print(a.join(b))

    # tuple = ("One", "Two", "Three")
    # value_str = " ".join(tuple)
    # print(value_str)

    # tuple_2 = ["The", "challenge", "is", "on"]
    # value_str_2 = "#".join(["The", "challenge", "is", "on"])
    # print(value_str_2)

    print(" ".join([Hello().why(), "adeus"]))

File: tests/expected/julia/join_test.jl
abstract type AbstractHello end
mutable struct Hello <: AbstractHello
end
function test(self::Hello)::String
    return "ola"
end

if abspath(PROGRAM_FILE) == @__FILE__
    a = "a"
    b = "ab"
    @assert(join(b, a) == "aab")
    tuple = ("One", "Two", "Three")
    value_str = join(tuple, " ")
    @assert(value_str == "One Two Three")
    tuple_2 = ["The", "challenge", "is", "on"]
    value_str_2 = join(["The", "challenge", "is", "on"], "#")
    @assert(value_str_2 == "The#challenge#is#on")
    @assert(join([test(Hello()), "adeus"], " ") == "ola adeus")
    @assert(join([test(Hello()), "adeus"], "\n") == "ola\nadeus")
    println("OK")
end
